src/core_modules/capture-core/components/WidgetEnrollment/WidgetEnrollment.container.js

```diff
@@ -22,11 +22,11 @@ const useContainsAutoGeneratedEvent = program =>
 
 const useCanAddNew = (enrollments, programId, tetAccess) =>
     useMemo(
-        () => tetAccess?.write &&
+        () => tetAccess?.data?.write &&
             enrollments
                 .filter(item => item.program === programId)
                 .every(item => item.status !== plainStatus.ACTIVE),
-        [enrollments, programId, tetAccess?.write],
+        [enrollments, programId, tetAccess?.data?.write],
```

Comment:
This `?.` notation expands into a complicated expression, and I'm not yet used to seeing complicated expressions in the dependency array. Can we store `tetAccess?.data?.write` in a variable so we 1. don't need to evaluate it twice and 2. keep me unused of seeing complicated expressions in the dependency array? (Maybe this turned into a dilemma when I wording my request this way..)

Reply:
Sure. Just removing everything behind `tetAccess` should have the same result either way, as there will never be partial updates of this object. Reverting to just `[enrollments, programId, tetAccess]`.